packages/react-dom-bindings/src/server/ReactDOMServerFormatConfig.js

```diff
@@ -2835,9 +2978,40 @@ const arraySubsequentOpenBracket = stringToPrecomputedChunk(',[');
 const arrayInterstitial = stringToPrecomputedChunk(',');
 const arrayCloseBracket = stringToPrecomputedChunk(']');
 
+type EmbeddingContext = 0 | 1;
+const HTMLAttributeEmbedding: EmbeddingContext = 0;
+const JavascriptEmbedding: EmbeddingContext = 1;
+
+function writeStyleResourceObject(
+  destination: Destination,
+  str: string,
+  context: EmbeddingContext,
+) {
+  if (context === JavascriptEmbedding) {
```

Comment:
Yeah, this felt pretty icky to write and reason about. Happy to change it to something else!

> It might be better to encode extra "props" for styles as additional data attributes instead of as JSON.

Would this look something like
```
data-resource1-href="..."
data-resource1-precedence="..."
data-resource1-prop1-key="..."
data-resource1-prop1-val="..."

data-resource2-href="..."
data-resource2-precedence="..."
data-resource2-prop1-key="..."
data-resource2-prop1-val="..."
```

From a bytes perspective, sending extra `&quot;`s might be cheaper than adding additional data attributes. Is the issue with JSON.parse, how modular the code is, or something else?

Reply:
Ah. I was thinking more like:

```
data-h="..."
data-p="..."
data-key="value"
data-key="value"
data-key="value"
```

I didn't think that there were multiple.

One option might be to actually use it as a template to rely on the HTML parsing:

```
<template...>
  <link rel="..." data-precedence="..." key="value" />
  <link rel="..." data-precedence="..." key="value" />
  <link rel="..." data-precedence="..." key="value" />
  ...
</template>
```

Might not be better though.

Reply:
Ah I see, so each `link` tag could encode a style resource (if I'm understanding)

Should this PR be changed to use a `template`/`link` format, or is the JSON encoding implementation alright for now?

Reply:
We can leave the JSON encoding for now but let's fork `writeStyleResourceDependencies` so that it's clear/easy that it can have a completely different format when we change it.